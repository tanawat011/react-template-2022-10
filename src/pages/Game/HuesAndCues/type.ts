/* istanbul ignore file */
import type { DocumentData, QueryDocumentSnapshot } from 'firebase/firestore'

export type Player = {
  id: string
  name: string
}

export type Room = {
  id: string
  password: string
  hintChoice: string[]
  hintSelected: string
  isStarted: boolean
  isSubmitResult: boolean
  totalRound: number
}

export type RawRoomPlayer = {
  refPlayer: QueryDocumentSnapshot<DocumentData>
  color: string
  score: number
  allSelected: string[]
  isOwner: boolean
  isTurn: boolean
  isHinter: boolean
  totalTurn: number
  seq: number
}

export type RoomPlayer = Omit<RawRoomPlayer, 'refPlayer'> & {
  player: Player
}
