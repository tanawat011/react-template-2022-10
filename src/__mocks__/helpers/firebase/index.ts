export const deleteCollection = jest.fn()
export const deleteDocument = jest.fn()
export const getDocument = jest.fn()
export const getAllDocuments = jest.fn()
export const setDocument = jest.fn()
export const subscribeCollection = jest.fn()
export const updateDocument = jest.fn()
