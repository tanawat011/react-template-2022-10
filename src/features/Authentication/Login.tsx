import { PATH } from 'Routes'
import { Button } from 'components/Button'

export const Login = () => {
  return (
    <>
      <div id='login-form'>
        <h1>Login</h1>

        <form className='flex flex-col'>
          <label id='username'>Username</label>
          <input id='username' />
          <label id='password'>Password</label>
          <input id='password' />

          <div className='flex'>
            <Button link to={`/${PATH.BACKOFFICE.HOME}`}>
              Login
            </Button>
            <Button link to={`../${PATH.AUTH.FORGOT_PASSWORD}`}>
              Forgot Password
            </Button>
          </div>
        </form>
      </div>
    </>
  )
}
