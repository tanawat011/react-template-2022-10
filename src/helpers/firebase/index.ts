export * from './deleteCollection'
export * from './deleteDocument'
export * from './getAllDocuments'
export * from './getDocument'
export * from './setDocument'
export * from './subscribeCollection'
export * from './updateDocument'
