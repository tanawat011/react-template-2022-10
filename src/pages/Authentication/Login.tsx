// import type { CustomEventTarget } from 'types/html'

import type { FormEvent } from 'react'

// import { getAuth, signInWithEmailAndPassword } from 'firebase/auth'
import { useNavigate } from 'react-router-dom'

import { PATH } from 'Routes'
import { Button } from 'components/Button'

export const Login = () => {
  // const auth = getAuth()

  const navigate = useNavigate()

  const submitLogin = async (e: FormEvent) => {
    e.preventDefault()

    navigate(`/${PATH.BACKOFFICE.HOME}`)

    // try {
    //   const target = e.target as CustomEventTarget<{
    //     username: { value: string }
    //     password: { value: string }
    //   }>

    //   const username = target.username.value
    //   const password = target.password.value

    //   const response = await signInWithEmailAndPassword(auth, username, password)

    //   if (response) {
    //     const session = {
    //       uid: response.user.uid,
    //       email: response.user.email,
    //     }

    //     sessionStorage.setItem('user', JSON.stringify(session))

    //     navigate(`/${PATH.BACKOFFICE.HOME}`)
    //   }
    // } catch (error) {
    //   // eslint-disable-next-line no-console
    //   console.log(error)
    // }
  }

  return (
    <>
      <div id='login-form'>
        <h1>Login</h1>

        <form className='flex flex-col' onSubmit={submitLogin}>
          <label id='username'>Username</label>
          <input id='username' />
          <label id='password'>Password</label>
          <input id='password' />

          <div className='flex'>
            <Button type='submit'>Login</Button>
            <Button link to={`../${PATH.AUTH.FORGOT_PASSWORD}`}>
              Forgot Password
            </Button>
          </div>
        </form>
      </div>
    </>
  )
}
