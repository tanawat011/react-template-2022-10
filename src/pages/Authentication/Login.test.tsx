import type { RenderResult } from '@testing-library/react'

import { act, fireEvent, screen, waitFor } from '@testing-library/react'

import { renderWithProviders } from 'helpers/test'

import { Login } from './Login'

jest.mock('components/Button', () => ({
  Button: ({ link, ...props }: React.HTMLAttributes<never> & { link: boolean }) => {
    console.log(link)

    return (
      <div {...props} data-testid={props.children}>
        {props.children}
      </div>
    )
  },
}))

describe('<Login />', () => {
  test('match snapshot', () => {
    const { asFragment } = renderWithProviders(<Login />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('submit form', async () => {
    let el: RenderResult

    await act(async () => {
      el = renderWithProviders(<Login />)
    })

    const button = screen.getByTestId(/Login/i)
    fireEvent.click(button)

    await waitFor(() => {
      expect(el.asFragment()).toBeTruthy()
    })
  })
})
