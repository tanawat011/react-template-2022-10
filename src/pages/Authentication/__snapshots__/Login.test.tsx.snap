// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> match snapshot 1`] = `
<DocumentFragment>
  <div
    id="login-form"
  >
    <h1>
      Login
    </h1>
    <form
      class="flex flex-col"
    >
      <label
        id="username"
      >
        Username
      </label>
      <input
        id="username"
      />
      <label
        id="password"
      >
        Password
      </label>
      <input
        id="password"
      />
      <div
        class="flex"
      >
        <div
          data-testid="Login"
          type="submit"
        >
          Login
        </div>
        <div
          data-testid="Forgot Password"
          to="../forgot-password"
        >
          Forgot Password
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
